src/relay/pass/feature.cc

```diff
@@ -34,17 +34,23 @@ namespace relay {
 
 FeatureSet DetectFeature(const Expr& expr) {
   if (!expr.defined()) {
-    return FeatureSet::NoFeature();
+    return FeatureSet::No();
   }
   struct FeatureDetector : ExprVisitor {
     std::unordered_set<Expr, NodeHash, NodeEqual> visited_;
-    FeatureSet fs = FeatureSet::NoFeature();
+    bool primitive_ = false;
+    FeatureSet fs = FeatureSet::No();
+
     void VisitExpr(const Expr& expr) final {
       if (visited_.count(expr) == 0) {
+        visited_.insert(expr);
         ExprVisitor::VisitExpr(expr);
       } else {
         if (!IsAtomic(expr)) {
           fs += fGraph;
+          if (!primitive_) {
```

Comment:
When would `primitive_` ever be set to true?

Reply:
thx! I only implemented half of this and totally forgot!